metsrw/metadata.py

```diff
@@ -298,14 +299,18 @@ class MDWrap(object):
         include "PREMIS:OBJECT", "PREMIS:EVENT,", "DC" and "OTHER".
     :param str othermdtype: The OTHERMDTYPE of the XML document. Should be set if mdtype is "OTHER".
     """
-    def __init__(self, document, mdtype, othermdtype=None):
+
+    MDTYPE_CLASSES = {'DC': DublinCoreXmlData}
+
+    def __init__(self, document, mdtype, othermdtype=None, data=None):
         parser = etree.XMLParser(remove_blank_text=True)
         if isinstance(document, six.string_types):
             self.document = etree.fromstring(document, parser=parser)
         elif isinstance(document, (etree._Element, list)):
             self.document = document
         self.mdtype = mdtype
         self.othermdtype = othermdtype
+        self.data = data
```

Comment:
Comment by @Hwesta: Rather than creating a new attribute, this could use document to store the child document - whether that's a string, ElementTree, or plugin class of the appropriate type.